framework/src/base/MaxQpsThread.C

```diff
@@ -63,26 +65,26 @@ MaxQpsThread::operator() (const ConstElemRange & range)
       // We cannot mess with the FE objects in Assembly, because we might need to request second derivatives
       // later on. If we used them, we'd call reinit on them, thus making the call to request second
       // derivatives harmful (i.e. leading to segfaults/asserts). Thus, we have to use a locally allocated object here.
-      FEBase * fe = FEBase::build(dim, fe_type).release();
+      UniquePtr<FEBase> fe (FEBase::build(dim, fe_type));
 
       // figure out the number of qps for volume
-      QBase * qrule = QBase::build(_qtype, dim, _order).release();
-      fe->attach_quadrature_rule(qrule);
+      UniquePtr<QBase> qrule (QBase::build(_qtype, dim, _order));
+      fe->attach_quadrature_rule(qrule.get());
```

Comment:
No, I'm not creating a UniquePtr(), you were by calling FEBase::build()!
http://libmesh.github.io/doxygen/classlibMesh_1_1FEGenericBase.html#ae11f16dfbb5b86d2d52ede7490e39f59

The only difference between your code and mine is that you explicitly throw it away after you build with `release()`. I just kept it that's all. We are both digging through the class to get the raw pointer, you are just doing it one line earlier.

After reading all of the Scott Meyer's books, I've really started to think about how I create pointer code. He gives several examples of how memory can be leaked even inside of a single basic block simply by the fact that an exception can occur after allocation, but before assignment so that it never gets released. It's fairly eye opening bug that can cause subtle issues in certain situations. Now I admit that this particular situation would cause the program to abort here, but it's actually far more difficult to see memory errors than you might expect. If that object had been allocated as a raw pointer, I probably wouldn't have changed it but there's no reason to throw away the smart pointer you are handed.